packages/react-components/src/components/utils/unit-controller-history-handler.ts

```diff
@@ -0,0 +1,84 @@
+import { TimelineChart } from 'timeline-chart/lib/time-graph-model';
+import { TimeGraphUnitController } from 'timeline-chart/lib/time-graph-unit-controller';
+
+export interface HistoryItem {
+    selectionRange: TimelineChart.TimeGraphRange | undefined;
+    viewRange: TimelineChart.TimeGraphRange;
+}
+
+export class UnitControllerHistoryHandler {
+    protected stack: Array<HistoryItem>;
+    protected i = 0;
+    protected max = 0;
+    protected timeout?: ReturnType<typeof setTimeout>;
+    protected unitController: TimeGraphUnitController;
+    protected restoring = false;
+
+    constructor(uC: TimeGraphUnitController) {
+        this.stack = [];
+        this.unitController = uC;
+    }
+
+    public addCurrentState(): void {
+        const { selectionRange, viewRange } = this.unitController;
+        this.waitToAdd({ selectionRange, viewRange });
+    }
```

Comment:
Currently only storing ViewRange and SelectionRange.  If we want anything else added please let me know!  However, it will require refactoring.

Reply:
I think it's a good start for the initial implementation. We can have subsequent PRs for more things if needed.